tests/test_pydantic.py

```diff
@@ -379,3 +380,20 @@ class PyType(PyTypeCustomBase):
         ],
     }
     assert_schema(PyType, expected)
+
+
+def test_field_by_alias():
```

Comment:
I know the current logic covers this, but could we add a test where the Pydantic field does NOT have an alias, but we still set the alias option?

Just to avoid someone breaking this in the future.